drivers/ethernet/eth_lan9250.c

```diff
@@ -0,0 +1,713 @@
+	 * Where phy_add = 0b00001 & index = address
+	 * Data = ((phy_add & 0x1F) << 11) | ((index & 0x1F) << 6)
+	 */
+	lan9250_write_mac_reg(dev, LAN9250_HMAC_MII_ACC, (1 << 11) | ((address & 0x1F) << 6));
+
+	/* Wait PHY to be ready and send reading register command */
+	lan9250_wait_mac_ready(dev, LAN9250_HMAC_MII_ACC, LAN9250_HMAC_MII_ACC_MIIBZY, 0,
+			       LAN9250_PHY_TIMEOUT);
+
+	/* Read 32bit value from the indirect MAC registers */
+	lan9250_read_mac_reg(dev, LAN9250_HMAC_MII_DATA, &tmp);
+	*value = tmp;
+
+	return 0;
+}
+
+static int lan9250_write_phy_reg(const struct device *dev, uint8_t address, uint16_t data)
+{
+	/* Wait PHY to be ready and send reading register command */
+	lan9250_wait_mac_ready(dev, LAN9250_HMAC_MII_ACC, LAN9250_HMAC_MII_ACC_MIIBZY, 0,
+			       LAN9250_PHY_TIMEOUT);
+	lan9250_write_mac_reg(dev, LAN9250_HMAC_MII_DATA, data);
+
+	/* Reference: Microchip Ethernet LAN9250
+	 * https://github.com/microchip-pic-avr-solutions/ethernet-lan9250/
+	 *
+	 * Datasheet:
+	 * https://ww1.microchip.com/downloads/aemDocuments/documents/OTH/ProductDocuments/DataSheets/00001913A.pdf
+	 *
+	 * 12.2.18 PHY REGISTERS
+	 * The PHY registers are indirectly accessed through the Host MAC MII Access Register
+	 * (HMAC_MII_ACC) and Host MAC MII Data Register (HMAC_MII_DATA).
+	 *
+	 * Write 32bit value to the indirect MAC registers
+	 * Where phy_add = 0b00001 & index = address
+	 * Data = ((phy_add & 0x1F) << 11) | ((index & 0x1F)<< 6) | MIIWnR
+	 */
+	lan9250_write_mac_reg(dev, LAN9250_HMAC_MII_ACC,
+			      (1 << 11) | ((address & 0x1F) << 6) | LAN9250_HMAC_MII_ACC_MIIW_R);
+
+	/* Wait PHY to be ready and send reading register command */
+	lan9250_wait_mac_ready(dev, LAN9250_HMAC_MII_ACC, LAN9250_HMAC_MII_ACC_MIIBZY, 0,
+			       LAN9250_PHY_TIMEOUT);
+
+	return 0;
+}
+
+static int lan9250_set_macaddr(const struct device *dev)
+{
+	struct lan9250_runtime *ctx = dev->data;
+
+	lan9250_write_mac_reg(dev, LAN9250_HMAC_ADDRL,
+			      ctx->mac_address[0] | (ctx->mac_address[1] << 8) |
+			      (ctx->mac_address[2] << 16) | (ctx->mac_address[3] << 24));
+	lan9250_write_mac_reg(dev, LAN9250_HMAC_ADDRH,
+			      ctx->mac_address[4] | (ctx->mac_address[5] << 8));
+
+	return 0;
+}
+
+static int lan9250_spi_check(const struct device *dev)
+{
+	uint32_t tmp;
+	uint8_t retries = LAN9250_DEFAULT_NUMOF_RETRIES;
+
+	do {
+		lan9250_read_sys_reg(dev, LAN9250_BYTE_TEST, &tmp);
+		k_busy_wait(USEC_PER_MSEC * 1U);
+	} while (tmp != LAN9250_BYTE_TEST_DEFAULT && retries--);
+
+	return 0;
+}
+
+static int lan9250_hw_cfg_check(const struct device *dev)
+{
+	uint32_t tmp;
+
+	do {
+		lan9250_read_sys_reg(dev, LAN9250_HW_CFG, &tmp);
+		k_busy_wait(USEC_PER_MSEC * 1U);
+	} while ((tmp & LAN9250_HW_CFG_DEVICE_READY) == 0);
+
+	return 0;
+}
+
+static int lan9250_sw_reset(const struct device *dev)
+{
+	/* Wait until LAN9250 SPI bus is ready */
+	lan9250_wait_ready(dev, LAN9250_BYTE_TEST, BOTR_MASK, LAN9250_BYTE_TEST_DEFAULT,
```

Comment:
Is this call necessary? The `lan9250_spi_check()` already does this and that is called before this function.

Reply:
You are right, it is redundant and it is not really needed